src/base/BaseStrategy.sol

```diff
@@ -0,0 +1,396 @@
+// SPDX-License-Identifier: BSD-3-Clause
+pragma solidity ^0.8.24;
+
+import {IERC20, Math, SafeERC20} from "src/Common.sol";
+import {Vault} from "src/Vault.sol";
+
+/**
+ * @title BaseStrategy
+ * @author Yieldnest
+ * @notice This contract is a base strategy for any underlying protocol.
+ * vault.
+ */
+abstract contract BaseStrategy is Vault {
```

Comment:
let's try an experiment here with external libraries to reduce code size of contract

Maybe a branch off of this branch is best?